lib/charms/tls_certificates_interface/v2/tls_certificates.py

```diff
@@ -984,6 +1018,40 @@ def generate_csr(
     return signed_certificate.public_bytes(serialization.Encoding.PEM)
 
 
+def csr_matches_certificate(csr: str, cert: str) -> bool:
+    """Check if a CSR matches a certificate.
+
+    expects to get the original string representations.
```

Comment:
This comment can be next to the `csr` args a couple of lines lower

Reply:
Done